code/modules/reagents/chemistry/machinery/reagentgrinder.dm

```diff
@@ -3,62 +3,84 @@
 	icon = 'icons/obj/kitchen.dmi'
 	icon_state = "juicer1"
 	layer = 2.9
-	density = 1
 	anchored = 1
 	use_power = 1
 	idle_power_usage = 5
 	active_power_usage = 100
-	var/inuse = 0
+	pass_flags = PASSTABLE
+	var/operating = 0
 	var/obj/item/weapon/reagent_containers/beaker = null
 	var/limit = 10
 
 	//IMPORTANT NOTE! A negative number is a multiplier, a positive number is a flat amount to add. 0 means equal to the amount of the original reagent
 	var/list/blend_items = list (
 
-		//Sheets
-		/obj/item/stack/sheet/mineral/plasma = list("plasma_dust" = 20),
-		/obj/item/stack/sheet/mineral/uranium = list("uranium" = 20),
-		/obj/item/stack/sheet/mineral/bananium = list("banana" = 20),
-		/obj/item/stack/sheet/mineral/tranquillite = list("nothing" = 20),
-		/obj/item/stack/sheet/mineral/silver = list("silver" = 20),
-		/obj/item/stack/sheet/mineral/gold = list("gold" = 20),
-		/obj/item/weapon/grown/novaflower = list("capsaicin" = 0),
-
-		//archaeology!
-		///obj/item/weapon/rocksliver = list("ground_rock" = 50),
-
-
-		//All types that you can put into the grinder to transfer the reagents to the beaker. !Put all recipes above this.!
-		/obj/item/weapon/reagent_containers/food = list(),
-		/obj/item/weapon/reagent_containers/honeycomb = list()
-	)
-
-	var/list/blend_tags = list (
-		"nettle" = list("sacid" = 0),
-		"deathnettle" = list("facid" = 0),
-		"soybeans" = list("soymilk" = 0),
-		"tomato" = list("ketchup" = 0),
-		"wheat" = list("flour" = -5),
-		"rice" = list("rice" = -5),
-		"cherries" = list("cherryjelly" = 0),
+			//Sheets
+			/obj/item/stack/sheet/mineral/plasma = list("plasma" = 20),
+			/obj/item/stack/sheet/metal = list("iron" = 20),
+			/obj/item/stack/sheet/plasteel = list("iron" = 20, "plasma" = 20),
+			/obj/item/stack/sheet/wood = list("carbon" = 20),
+			/obj/item/stack/sheet/glass = list("silicon" = 20),
+			/obj/item/stack/sheet/rglass = list("silicon" = 20, "iron" = 20),
+			/obj/item/stack/sheet/mineral/uranium = list("uranium" = 20),
+			/obj/item/stack/sheet/mineral/bananium = list("banana" = 20),
+			/obj/item/stack/sheet/mineral/tranquillite = list("nothing" = 20),
+			/obj/item/stack/sheet/mineral/silver = list("silver" = 20),
+			/obj/item/stack/sheet/mineral/gold = list("gold" = 20),
```

Comment:
Rods (from the steel cap) are not grindable. Is this deliberate, so as to require processing before being able to get elemental iron?